synphot/models.py

```diff
@@ -185,7 +185,18 @@ class Box1D(_models.Box1D):
     """Same as `astropy.modeling.functional_models.Box1D`, except with
     ``sampleset`` defined.
 
+    Parameters
+    ----------
+    step : float
+        Distance of first and last points w.r.t. bounding box. In default units
+        (nominally Angstrom). Defaults to 0.01
+
     """
+    def __init__(self, step=0.01, *args, **kwargs):
```

Comment:
This will break API. I wonder if it is safer to extract `step` from `kwargs`.

Reply:
Might also have to register it here

https://github.com/spacetelescope/synphot_refactor/blob/a5fa6533bdcb58bb43d8ee43526d15c9817292bf/synphot/spectrum.py#L74

```python
'Box1D': {'amplitude': 'flux', 'x_0': 'wave', 'width': 'wave', 'step': 'wave'},
```

Reply:
Sorry - could you please point me to what exactly in the API will break? keywords should automatically be popped from **kwargs at runtime.  There should be no distinction between ``Box1D(**{'amplitude':1, 'x_0':1, 'width':1, 'step':1})`` and ``Box1D(amplitude=1, x_0 = 1, width=1, step=1)`` and any combination of separate keywords/kw dict.

Reply:
Since `Box1D` signature does accept `*args`, I am worried the first positional argument might be passed in as `step` with this change.

Reply:
Ah. You're absolutely right.  I'll re-implement as you suggest. Thanks.